include/data_struct.h

```diff
@@ -134,11 +134,28 @@ class BaseDeltaStates {
     virtual void swap(BaseDeltaStates &other);
 };
 
+class BaseLinearSmoother {
+   public:
+    float mu_prior, var_prior;
+    float mu_post, var_post;
+    std::vector<float> mu_w, var_w, var_b;
+    std::vector<float> mu_zo_smooths, var_zo_smooths;
+    size_t num_w = 0;
+
+    BaseLinearSmoother(size_t num_w);
+    BaseLinearSmoother();
+    ~BaseLinearSmoother() = default;
+    // virtual void set_size(size_t size, size_t size_w);
+    virtual void set_num_states_w(size_t num_w);
+    void reset_zeros();
+};
+
```

Comment:
yes. I will.

Reply:
I could not move this to the end, because the `BaseTempStates` uses `BaseTempSLinear` so that `BaseTempSLinear` need to be defined before. I also tried it to the end and using forward declaration, but it gave me errors that I could not solve.